test-suite-data.json

```diff
@@ -47,6 +47,30 @@
     "subpath": "googleapis/api/annotations",
     "is_invalid": false
   },
+  {
+    "description": "invalid subpath - unencoded subpath cannot contain '..'",
+    "purl": "pkg:GOLANG/google.golang.org/genproto@abcdedf#/googleapis/%2E%2E/api/annotations/",
+    "canonical_purl": "pkg:golang/google.golang.org/genproto@abcdedf#googleapis/api/annotations",
+    "type": "golang",
+    "namespace": "google.golang.org",
+    "name": "genproto",
+    "version": "abcdedf",
+    "qualifiers": null,
+    "subpath": "googleapis/../api/annotations",
```

Comment:
Should this be with the `..` or without? If we're using the value as input to verify that formatting removes the `..`, then we want `..` here, but if we're verifying that parsing removes the `..` we wouldn't want it here.

Reply:
> Thinking of some other test objects I've seen, must a canonical_purl be either null or a valid purl? Or another way to think about it: when is a canonical_purl null, and can it ever be an invalid purl (like https://github.com/package-url/purl-spec/blob/main/test-suite-data.json#L197)?


I have no idea.lets create a ticket for this :)